File: linked_list_demo.cpp
#include "./ds_lib/cpp/LinkedList.h"
#include <iostream>

using namespace std;

int main() {
  LinkedList<char>* myLL = new LinkedList<char>();

  myLL->insertAtHead('b');
  myLL->insertAtEnd('c');
  myLL->insertAtEnd('d');
  myLL->insertAtEnd('e');
  myLL->insertAtEnd('f');
  myLL->insertAtHead('a');
  
  cout << myLL->toString() << endl;

  myLL->deleteAtEnd();
  myLL->remove('b');
  myLL->remove('z');
  myLL->deleteAtHead();
  Node<char>* found = myLL->search('c');

  cout << found->data << endl;
  
  cout << myLL->toString() << endl; // expected output: c->d->e
  cout << myLL->size() << endl;
}

File: LinkedListDemo.java
import ds_lib.java.CustomLinkedList;

public class LinkedListDemo {
  public static void main(String[] args) {
    CustomLinkedList<String> myLL = new CustomLinkedList<>();

    myLL.insertAtEnd("ball");
    myLL.insertAtEnd("cat");
    myLL.insertAtEnd("dog");
    myLL.insertAtEnd("elephant");
    myLL.insertAtEnd("fish");
    myLL.insertAtHead("apple");

    System.out.println(myLL.toString());

    myLL.remove("fish");
    System.out.println(myLL.size());
    System.out.println(myLL.toString());
  }
}
